### PR TITLE
Dump Models and dependencies in core.py work

### PR DESCRIPTION
**Pull request by octaviansima**

### CODE REVIEW

**Review by podcastinator — CHANGES_REQUESTED** [review]
Excellent, thanks! Looks good for the most part! Please fix the following?
1. Left a few review comments in the code: some nits, and a couple of security enhancements.
2. Merge in the latest commits on master from the main repo  into your branch, and make sure your changes still build.
3. Also, add your changes to the demo file that you used for testing?

File: enclave/xgboost.conf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Enclave settings:
 Debug=1
-NumHeapPages=100000
+NumHeapPages=20000
```
Comment (chester-leung) [inline enclave/xgboost.conf]: Can you revert this please?

**Comment by podcastinator** [comment]
LGTM, thanks! @chester-leung Can you double-check to make sure this won't break the remote-control demo?

**Review by podcastinator — APPROVED** [review]

**Comment by chester-leung** [comment]
@podcastinator will do. We'll have to wait for ACC access though before we can test anything and know for sure.

**Comment by chester-leung** [comment]
We'll have to add another RPC call for `get_fscore()` to return the output from the server to the client. We can take care of this in another PR though.

File: enclave/src/c_api/c_api.cc [hunk 2]
```diff
@@ -16,6 +16,7 @@
 #include <vector>
 #include <string>
 #include <memory>
+#include "base64.h"
```
Comment (chester-leung) [inline enclave/src/c_api/c_api.cc]: Should this be `#include <dmlc/base64.h>`? I had to change it to that to build the code.

If so could you please move this line to under line 9, `#include <dmlc/thread_local.h>`? Thanks!

**Comment by chester-leung** [comment]
LGTM thanks!

**Pull request merged**